build.sh

```diff
@@ -175,27 +186,66 @@ if [ "$MACHINE" == "ncar" ]; then
   fi
 fi
 
+# comma-separated list of files in src/framework that are needed to build $LININFRA (for FMS2, at least)
+MOM6_infra_framework_deps_list=$(cat << EOF
+MOM_string_functions.F90
+MOM_io.F90
+MOM_array_transform.F90
+MOM_domains.F90
+MOM_error_handler.F90
+posix.F90
+MOM_file_parser.F90
+MOM_coms.F90
+MOM_document.F90
+MOM_cpu_clock.F90
+MOM_unit_scaling.F90
+MOM_dyn_horgrid.F90
+MOM_hor_index.F90
+MOM_ensemble_manager.F90
+MOM_io_file.F90
+MOM_netcdf.F90
+EOF
+)
+MOM6_infra_framework_deps=$(echo ${MOM6_infra_framework_deps_list} | tr ' ' ',')
+# comma-separated list of files in src/core that are needed to build $LIBINFRA (for FMS2, at least)
+MOM6_infra_core_deps=MOM_grid.F90,MOM_verticalGrid.F90
+MOM6_infra_files=${MOM_ROOT}/{config_src/memory/${MEMORY_MODE},config_src/infra/${INFRA},src/framework/{$MOM6_infra_framework_deps},src/core/{$MOM6_infra_core_deps}}
+MOM6_src_files=${MOM_ROOT}/{config_src/memory/${MEMORY_MODE},config_src/drivers/solo_driver,pkg/CVMix-src/src/shared,pkg/GSW-Fortran/modules,../MARBL/src,config_src/external,src/{*,*/*}}/
+
+# 1) Build Underlying Infrastructure Library
+if [ "${INFRA}" == "FMS2" ]; then
+  cd ${BLD_PATH}
+  mkdir -p FMS
+  cd FMS
+  ${MKMF_ROOT}/list_paths ${FMS_ROOT}
+  # We need shr_const_mod.F90 and shr_kind_mod.F90 from ${SHR_ROOT}/src to build FMS
+  echo "${SHR_ROOT}/src/shr_kind_mod.F90" >> path_names
+  echo "${SHR_ROOT}/src/shr_const_mod.F90" >> path_names
+  ${MKMF_ROOT}/mkmf -t ${TEMPLATE} -p libfms.a -c "-Duse_libMPI -Duse_netCDF -DSPMD" path_names
+  make -j${JOBS} DEBUG=${DEBUG} CODECOV=${CODECOV} OFFLOAD=${OFFLOAD} libfms.a
+  LINKING_FLAGS="-L../MOM6-infra -linfra-${INFRA} -L../FMS -lfms"
```

Comment:
I suggest adding an else clause to throw an error if an unsupported INFRA option is specified. (If I am following the changes correctly, anything other than "FMS2" should lead to an error, until we introduce more option(s).)

Reply:
That's a great idea, I'll add a check for that